File: topics/control-flow/exercises/draw_bar_pattern.c
void draw_bar_pattern(int num_bars)
{
    int x, i, bar_width;
    
    i = 0;
    bar_width = screen_width() / num_bars;
    
    while( i < num_bars )
    {
        x = i * bar_width;
        if ( (i % 2) == 0 )
            fill_rectangle(ColorWhite, x, 0, bar_width, screen_height());
        else
            fill_rectangle(ColorBlack, x, 0, bar_width, screen_height());
        
        i++;
    }
    refresh_screen();
}

File: topics/control-flow/exercises/DrawBarPattern.pas
procedure DrawBarPattern(numBars: Integer);
var
    x, i, barWidth: Integer;
begin
    i := 0;
    barWidth := ScreenWidth() / numBars;
    
    while i < numBars do
    begin
        x := i * barWidth;
        if (i mod 2) = 0 then
            FillRectangle(ColorWhite, x, 0, barWidth, ScreenHeight())
        else
            FillRectangle(ColorBlack, x, 0, barWidth, ScreenHeight());
        
        i += 1;
    end;
    
    RefreshScreen();
end